pkg/test/framework/components/apps/native.go

```diff
@@ -357,27 +375,24 @@ func (a *nativeApp) Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedResp
 		return nil, err
 	}
 
-	if len(resp) != 1 {
-		return nil, fmt.Errorf("unexpected number of responses: %d", len(resp))
-	}
-	if !resp[0].IsOK() {
-		return nil, fmt.Errorf("unexpected response status code: %s", resp[0].Code)
-	}
-	if resp[0].Host != dstServiceName {
-		return nil, fmt.Errorf("unexpected host: %s", resp[0].Host)
-	}
-	if resp[0].Port != strconv.Itoa(dst.port.ApplicationPort) {
-		return nil, fmt.Errorf("unexpected port: %s", resp[0].Port)
-	}
-
 	return resp, nil
 }
 
-func (a *nativeApp) CallOrFail(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse {
+func (a *nativeApp) ValidatedCall(e AppEndpoint, opts AppCallOptions, t testing.TB) []*echo.ParsedResponse {
 	r, err := a.Call(e, opts)
 	if err != nil {
 		t.Fatal(err)
```

Comment:
I'd imagine there could be a test where if a request fails due to some reason, e.g. network, then the test may want to try again, so maybe let the caller decide to stop the test if they want to, and change the function name to something like `GetResponseOrError` or `CallOrError` and return `([]*echo.ParsedRepsonse, error)`?